includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -2511,15 +2607,66 @@ private function finalize_styles() {
 					)
 				);
 			} else {
-				$css  = implode( '', $stylesheet_sets['keyframes']['final_stylesheets'] );
-				$css .= $stylesheet_sets['keyframes']['source_map_comment'];
+				$css  = implode(
+					'',
+					wp_list_pluck(
+						array_filter(
+							$this->pending_stylesheets,
+							function( $pending_stylesheet ) {
+								return $pending_stylesheet['included'] && 'keyframes' === $pending_stylesheet['group'];
+							}
+						),
+						'stylesheet'
+					)
+				);
+				$css .= $stylesheet_groups['keyframes']['source_map_comment'];
 
 				$style_element = $this->dom->createElement( 'style' );
 				$style_element->setAttribute( 'amp-keyframes', '' );
 				$style_element->appendChild( $this->dom->createTextNode( $css ) );
 				$body->appendChild( $style_element );
 			}
 		}
+
+		$this->remove_admin_bar_if_css_excluded();
+	}
+
+	/**
+	 * Remove admin bar if its CSS was excluded.
+	 *
+	 * @since 1.2
+	 */
+	private function remove_admin_bar_if_css_excluded() {
+		if ( ! is_admin_bar_showing() ) {
+			return;
+		}
+
+		$admin_bar = $this->dom->getElementById( 'wpadminbar' );
+		if ( ! $admin_bar ) {
+			return;
+		}
+
+		$included = true;
+		foreach ( $this->pending_stylesheets as &$pending_stylesheet ) {
+			$is_admin_bar_css = (
+				'custom' === $pending_stylesheet['group']
+				&&
+				$pending_stylesheet['node'] instanceof DOMElement
+				&&
+				'admin-bar-css' === $pending_stylesheet['node']->getAttribute( 'id' )
+			);
+			if ( $is_admin_bar_css ) {
+				$included = $pending_stylesheet['included'];
+				break;
+			}
+		}
+
+		if ( ! $included ) {
+			$admin_bar->parentNode->replaceChild(
+				$this->dom->createComment( ' ' . __( 'Admin bar (#wpadminbar) was removed to preserve AMP validity due to excessive CSS.', 'amp' ) . ' ' ),
```

Comment:
This should use a placeholder for `#wpadminbar` and a proper translator comment.

Reply:
Fixed in 08356df89c13ef4fc33bf3e7c6c06c517414772e.